tests/eosio.system_tester.hpp

```diff
@@ -775,22 +776,35 @@ class eosio_system_tester : public TESTER {
       return push_action( name(owner), "cnclrexorder"_n, mvo()("owner", owner) );
    }
 
-   action_result rentcpu( const account_name& from, const account_name& receiver, const asset& payment, const asset& fund = core_sym::from_string("0.0000") ) {
-      return push_action( name(from), "rentcpu"_n, mvo()
+   action_result rentcpu( const account_name& from, const account_name& receiver, const asset& payment, const bool channel_to_rex, const asset& fund = core_sym::from_string("0.0000") ) {
+      const auto act_ret =  push_action( name(from), "rentcpu"_n, mvo()
                           ("from",         from)
                           ("receiver",     receiver)
                           ("loan_payment", payment)
                           ("loan_fund",    fund)
       );
+
+      if(channel_to_rex) {
+         donatetorex("eosio.fee"_n, get_balance("eosio.fee"_n), "rentcpu");
+      }
+
+      return act_ret;
    }
 
-   action_result rentnet( const account_name& from, const account_name& receiver, const asset& payment, const asset& fund = core_sym::from_string("0.0000") ) {
-      return push_action( name(from), "rentnet"_n, mvo()
+   action_result rentnet( const account_name& from, const account_name& receiver, const asset& payment, const bool channel_to_rex, const asset& fund = core_sym::from_string("0.0000") ) {
+      const auto act_ret = push_action( name(from), "rentnet"_n, mvo()
                           ("from",         from)
                           ("receiver",     receiver)
                           ("loan_payment", payment)
                           ("loan_fund",    fund)
       );
+
+      if(channel_to_rex) {
```

Comment:
I do understand this direction

but perhaps it would be best to implement a basic `eosio.fees` smart contract that simply listens on `eosio.token::EOS` token transfer and donates to REX.

That way all the unit tests related to REX fees should technically all pass with minimal changes